dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyPath.java

```diff
@@ -30,16 +30,29 @@
 public final class DependencyPath {
 
   private List<Artifact> path = new ArrayList<>();
+  private List<String> scopes = new ArrayList<>();
+  private List<Boolean> optionals = new ArrayList<>();
```

Comment:
These two lines are the purpose of this PR; DependencyPath to hold scope and optional flag.

Reply:
so we have three parallel lists? That feels off. We should have one list of a class that contains artifact, scope, and optional.

Reply:
That's a nice idea. Updated with DependencyPath.Node class.